searx/plugins/hostnames.py

```diff
@@ -0,0 +1,107 @@
+# SPDX-License-Identifier: AGPL-3.0-or-later
+# pylint: disable=missing-module-docstring, too-many-branches
+
+import re
+from urllib.parse import urlunparse, urlparse
+
+from flask_babel import gettext
+
+from searx import settings
+from searx.plugins import logger
+from searx.settings_loader import get_yaml_file
+
+name = gettext('Hostnames plugin')
+description = gettext('Rewrite hostnames, remove results or prioritize them based on the hostname')
+default_on = False
+preference_section = 'general'
+
+plugin_id = 'hostnames'
+
+logger = logger.getChild(plugin_id)
+parsed = 'parsed_url'
+_url_fields = ['iframe_src', 'audio_src']
+
+
+def _load_regular_expressions(settings_key):
+    setting_value = settings.get(plugin_id, {}).get(settings_key)
+
+    if not setting_value:
+        return {}
+
+    # load external file with configuration
+    if isinstance(setting_value, str):
+        setting_value = get_yaml_file(setting_value)
+
+    if isinstance(setting_value, list):
+        return {re.compile(r) for r in setting_value}
+
+    if isinstance(setting_value, dict):
+        return {re.compile(p): r for (p, r) in setting_value.items()}
+
+    return {}
+
+
+# compatibility fallback for old hostname replace plugin
+# TODO: remove in the future once most/all instance maintainers finished migrating
+def _load_regular_expressions_with_fallback(settings_key):
+    expressions = _load_regular_expressions(settings_key)
+    if expressions:
+        return expressions
+
+    # fallback to the old `hostname_replace` settings format
+    hostname_replace_config = settings.get('hostname_replace', {})
+
+    if settings_key == 'replace':
+        return {re.compile(p): r for (p, r) in hostname_replace_config.items() if r != False}
+    else:
+        return {re.compile(p) for (p, r) in hostname_replace_config.items() if r == False}
```

Comment:
Following pylint here will also remove things like
- `www\.google\.de`
instead of what it should look like with the old `hostnames_replace` plugin
- `www.\google\.de: False`
but I guess that's okay